docs/developer_guide.md

```diff
@@ -3,6 +3,14 @@
 Each sub-project has its own developer guide. If you would like to work across
 projects, these instructions should help you get started:
 
+
+### Prepare your system
+
+(these instructions are tested on Ubuntu 24.04 which has clang-18 and python3.12 by default)
+
+```
+sudo apt update && sudo apt install -y python-is-python3 python3.12-venv python3-dev clang lld
```

Comment:
Might want to recommend https://github.com/pyenv/pyenv over `python-is-python3 python3.12-venv python3-dev`

I regularly switch between 3.10, 3.11, 3.12, 3.13, and 3.13t when testing packages.

Reply:
That's fine, but I wouldn't over-index on Ubuntu 24.04 and Python 3.12.

Reply:
Personally, I like just linking to pyenv and saying it's a good tool to use for managing multiple python environments. Probably too verbose for our docs, but if you wanted to check it out, here are some setup instructions:

## Pre-Reqs for ubuntu from their [wiki](https://github.com/pyenv/pyenv/wiki)
```bash
sudo apt update; sudo apt install build-essential libssl-dev zlib1g-dev \
libbz2-dev libreadline-dev libsqlite3-dev curl git \
libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
```

## Installing pyenv from instructions [here](https://github.com/pyenv/pyenv?tab=readme-ov-file#installation)
```bash
curl https://pyenv.run | bash
```

## Add to .bashrc
It's also outputted by installer after running installation:
```bash
echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc
echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
echo 'eval "$(pyenv init -)"' >> ~/.bashrc
```

## Run source ~/.bashrc
```bash
source ~/.bashrc
```

## Check installation
```bash
pyenv --version
```

## Install whichever version you need (i.e. for 3.12)
```bash
pyenv install 3.12
```

## Set as local version
In your workspace:
```bash
pyenv local 3.12
```

## Check python version
```bash
python --version
```

Reply:
yup i saw those. Made this based on that:

https://github.com/nod-ai/shark-ai/pull/585

Turns out pyenv can be just an extra step you do to replace python3-venv

I think we should just do venv for simplicity's sake but i'll add a lil click-expand section on using pyenv

Reply:
> Turns out pyenv can be just an extra step you do to replace python3-venv
> 
> I think we should just do venv for simplicity's sake but i'll add a lil click-expand section on using pyenv

pyenv and venv are different. Prior discussion in this repo: https://github.com/nod-ai/shark-ai/pull/524#discussion_r1842970285

Reply:
Sorry. mybad. not replace, but supplement.

Repalces python-is-python3, and docs adjusted accordingly.